tests/integration/bond_test.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+import copy
+
+import pytest
+import yaml
+
+from libnmstate import netapplier
+
+from .testlib import assertlib
+from .testlib import statelib
+from .testlib.statelib import INTERFACES
+
+
+BOND99_YAML_BASE = """
+interfaces:
+- name: bond99
+  type: bond
+  state: up
+  link-aggregation:
+    mode: balance-rr
+    slaves:
+    - eth1
+    - eth2
+"""
+
+
+@pytest.fixture
+def bond99_base():
+    state = yaml.load(BOND99_YAML_BASE)
+    netapplier.apply(copy.deepcopy(state))
+
+    yield copy.deepcopy(state)
+
+    state[INTERFACES][0]['state'] = 'absent'
```

Comment:
IMHO removing the bond should be tested explicitly in a test and not only in a fixture.

Reply:
Good point, thanks.